pkg/workceptor/kubernetes.go

```diff
@@ -430,23 +417,41 @@ func (kw *kubeUnit) runWorkUsingLogger() {
 
 			logStream, err := logReq.Stream(kw.ctx)
 			if err != nil {
-				errMsg := fmt.Sprintf("Error opening pod %s/%s stream: %s", ked.KubeNamespace, ked.PodName, err)
+				errMsg := fmt.Sprintf("Error opening pod %s/%s stream: %s", ked.KubeNamespace, kw.pod.Name, err)
 				kw.UpdateBasicStatus(WorkStateFailed, errMsg, 0)
 				logger.Error(errMsg)
 				break
 			}
 
 			streamReader := bufio.NewReader(logStream)
 
-			for errStdin != nil { //check between every line read to see if i need to stop reading
+			for { //check between every line read to see if i need to stop reading
 				line, err := streamReader.ReadString('\n')
+				errStdout = err
```

Comment:
This was what was making the tests fail. This used to be in the `io.Copy` line that we removed.